### DISCUSSION TITLE
Why doesn't the same seed produce identical images?

### DISCUSSION THREAD

**Discussion by halr9000** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#178] [Q&A] [answered]
I've been experimenting with some of the settings and found something odd. Have a look at the below. One base prompt, one negative, 3 regions identical size, using the same seed. I've tried a few variants on this and always, the first image is quite a bit different than the 2nd and 3rd, both of which are identical, or extremely similar. 

Correction: I forgot that I made the regions out of order! So instead, the regions 1 and 2 are identical (or close to it), and it's the 3rd which is not acting like it was using the same seed at all. Perhaps the first two aren't exactly identical due to some precision errors in the x/y dimensions? That would not explain the 3rd (left-most) here.

Any idea why? Seems a bit buggy.

![image](https://user-images.githubusercontent.com/18482/236658565-9ae3099d-d5f5-4242-8651-3215d39ce732.png)

> best quality, 4k, ultra realistic photo
> Negative prompt: 16-token-negative-deliberate-neg,
> Steps: 20, Sampler: DPM++ 2M Karras, CFG scale: 7, Seed: 1010549515, Size: 910x512, Model hash: 6bc8f9b865, Model: sd15_visiongenRealism_visiongenRealism, Tiled Diffusion: "{'Method': 'MultiDiffusion', 'Latent tile width': 96, 'Latent tile height': 96, 'Overlap': 48, 'Tile batch size': 1, 'Region control': {'Region 1': {'enable': True, 'x': 0.6666, 'y': 0, 'w': 0.3333, 'h': 1, 'prompt': 'ufo', 'neg_prompt': '', 'blend_mode': 'Foreground', 'feather_ratio': 0, 'seed': 243223430}, 'Region 2': {'enable': True, 'x': 0.3333, 'y': 0, 'w': 0.3333, 'h': 1, 'prompt': 'ufo', 'neg_prompt': '', 'blend_mode': 'Foreground', 'feather_ratio': 0, 'seed': 243223430}, 'Region 3': {'enable': True, 'x': 0, 'y': 0, 'w': 0.3333, 'h': 1, 'prompt': 'ufo', 'neg_prompt': '', 'blend_mode': 'Foreground', 'feather_ratio': 0, 'seed': 243223432}}}"

**Comment by Kahsolt** [comment, marked as answer]
As the footprint says, Region 1 & 2 uses seed **243223430**, but for Region 3 is **243223432**.
However why aren't the identical... 🤔 might be that canvas size is not dividable by 8, value precision error, or you enabled `draw_background`, or some unknown dark mystical power...

**Reply by halr9000** [reply]
Weird, the UI showed the same seed. I'll try to repro and if I can, open a bug. Otherwise, must be my mystical powers...
